Sources/TensorFlow/Core/BroadcastPullback.swift

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Pullback for broadcasting.
+public struct BroadcastPullback {
+  /// Constructs the pullback from broadcasting lhs and rhs.
+  public init<T: TensorFlowFloatingPoint, U: TensorFlowFloatingPoint>(
+    _ lhs: Tensor<T>, _ rhs: Tensor<U>
+  ) {
+    lhsShape = lhs.shape.dimensions.map { Int64($0) }
+    rhsShape = rhs.shape.dimensions.map { Int64($0) }
+  }
+
+  public func callAsFunction<T: TensorFlowFloatingPoint, U: TensorFlowFloatingPoint>(
+    _ lhsGrad: Tensor<T>, _ rhsGrad: Tensor<U>
+  ) -> (Tensor<T>, Tensor<U>) {
+    if lhsShape == rhsShape { return (lhsGrad, rhsGrad) }
+    let (lhsAxes, rhsAxes) = BroadcastPullback.computeAxes(lhsShape, rhsShape)
+    return (
+      _Raw.reshape(_Raw.sum(lhsGrad, reductionIndices: lhsAxes, keepDims: false), shape: lhsShape),
+      _Raw.reshape(_Raw.sum(rhsGrad, reductionIndices: rhsAxes, keepDims: false), shape: rhsShape)
+    )
+  }
+
+  /// Compute the axis needed to sum along in order to map back from the
+  /// broadcasted shape to the individual argument shapes
+  @usableFromInline
+  static func computeAxes(_ lhsShape: [Int64], _ rhsShape: [Int64]) -> ([Int64], [Int64]) {
```

Comment:
https://swift.org/documentation/api-design-guidelines/#label-closure-parameters

> * **Label tuple members and name closure parameters** where they appear in your API.
> 
> These names have explanatory power, can be referenced from documentation comments, and provide expressive access to tuple members.

This is one of the guidelines that I personally follow and recommend more like a rule (always apply it).

```suggestion
  static func computeAxes(_ lhsShape: [Int64], _ rhsShape: [Int64]) -> (lhsAxes: [Int64], rhsAxes: [Int64]) {
```